lib/src/signed_video_tlv.c

```diff
@@ -277,8 +277,10 @@ encode_general(signed_video_t *self, uint8_t *data)
     write_byte(last_two_bytes, &data_ptr, (uint8_t)((timestamp >> 8) & 0x000000ff), epb);
     write_byte(last_two_bytes, &data_ptr, (uint8_t)((timestamp)&0x000000ff), epb);
   }
-  for (size_t i = 0; i < gop_info->hash_size; i++) {
-    write_byte(last_two_bytes, &data_ptr, gop_info->hash_of_nalu_hash_list[i], epb);
+  if(!self->gop_hash_off) {
```

Comment:
As far as I can see in the code `gop_hash_off` is always `false`, hence you always enter this if-statement. May I suggest that you remove the if-statement since it adds nothing to the code.